### PR TITLE
Update rust grammar

### LINKED ISSUE #1985: Rust grammar doesn't appear to support all raw strings
The rules for raw strings in the rust grammar are as follows:

```antlr
RAW_STRING_LITERAL: 'r' RAW_STRING_CONTENT;
fragment RAW_STRING_CONTENT: '#' RAW_STRING_CONTENT '#' | '#"' .*? '"#';
```

This looks like it requires at least one pair of hashes (`#`) surrounding the quoted part. However, the [rust reference](https://doc.rust-lang.org/reference/tokens.html#characters-and-strings) specifies 0 or more `#` sets.

### PR DESCRIPTION
**Pull request by studentmain**
add new features, updated to rust v1.49.0 stable

close #1985

### CODE REVIEW

File: rust/README.md [hunk 1]
```diff
@@ -2,8 +2,13 @@
...
-This grammar only implemented v2018+ stable feature at the moment, not exactly match with official grammar (may differ at some edge case like isolated \r).
-
 Licensed under MIT
 
 Entry rule is `crate`, you need run test via command line at the moment: `grun Rust crate examples/*.rs`.
+
+Last updated for rust v1.49.0
+
+## Known limitation
+- Only v2018+ stable feature is implemented.
```
Comment (KvanTTT) [inline rust/README.md:12]: Please always insert a line break between header and paragraph, see [MD022 - Headers should be surrounded by blank lines](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines) rule.

```suggestion
## Known limitation

- Only v2018+ stable feature is implemented.
```
